subsys/nrf_security/src/drivers/cracen/cracenpsa/src/lib_kmu.c

```diff
@@ -34,6 +38,36 @@ void lib_kmu_clear_all_events(void)
 #endif
 }
 
+#if defined(CONFIG_SOC_SERIES_NRF71X)
+static void mram_sicr_perm_set(bool permission)
+{
+	if (permission) {
+		nrfx_mramc_config_write_mode_set(NRF_MRAMC_MODE_WRITE_DIRECT);
+		nrf_mramc_readynext_timeout_t readynext_timeout = {
+			.value		= NRF_MRAMC_READYNEXTTIMEOUT_DEFAULT,
+			.direct_write = true,
+		};
+		nrfy_mramc_readynext_timeout_set(NRF_MRAMC, &readynext_timeout);
+
+		/* CONFIGNVR.PAGE[3] page 4 contain the SICR info block
+		 * setting info block read and write permission to
+		 * allow KMU to write to SICR in MRAM.
+		 */
+		nrfy_mramc_confignvr_perm_set(true, 3);
```

Comment:
Much of the code in the `true` and `false` branches could be common.
```suggestion
		nrf_mramc_readynext_timeout_t readynext_timeout = {
			.value        = NRF_MRAMC_READYNEXTTIMEOUT_DEFAULT,
			.direct_write = permission,
		};
		nrfy_mramc_readynext_timeout_set(NRF_MRAMC, &readynext_timeout);

		/* CONFIGNVR.PAGE[3] page 4 contain the SICR info block
		 * setting info block read and write permission to
		 * allow KMU to write to SICR in MRAM.
		 */
		nrfy_mramc_confignvr_perm_set(permission, 3);
```

Reply:
updated.